deeplink testing

lordlamee opened the pull request:
So there are a few guidelines to testing this thing.
I've set the initial route to a screen where you can generate a deeplink..Longpress the text and you'll be able to copy the link..I know it's long but I should figure the short link very soon.
then change the initial route back to the loading screen
then hot restart or rerun the app.
if you click the deeplink , it should take you to the get started screen

NB. If you want the link to take you to other screens, set the route to the screen you want on main.dart, then go to the dynamic link test screen, then change the string passed into the createDynamicLinks function

I hope this works,

Other functionality would be added to the deeplink once the app is almost complete

lordlamee committed:
deeplink testing

Meghatronics reviewed (approved):
Great work OG!
Everything is here.

Meghatronics committed:
my tweaks

bamiogunfemi reviewed (approved):
LGTM

Meghatronics committed:
revert my tweaks

Meghatronics committed:
Changed long link to short link.
Created variable to hold the deeplink data.
Migrated link generation screen to Test page.
Handled incoming Links in main.dart.
Set logged in user in checkForUser instead of loadingAnimation.
Removed all new warnings to leave TODO's

Meghatronics commented:
App initial route has been returned to normal. 

+ Generate dynamic link from the test page.
+ Copy the link by a long press on it.
+ Close the app
+ Paste anywhere and click the link.
+ App opens. Navigate back to the test screen.
     You should see the data passed into the link display.
